src/gui/BandSplitter/TriStateButtonAttachment.cpp

```diff
@@ -38,21 +46,21 @@ void TriStateButtonAttachment::buttonClicked (juce::Button* button)
 
     if (currentState.first == BandState::TwoBands)
     {
-        currentState = std::make_pair (BandState::ThreeBands, 3);
         threeBandParam.setValueNotifyingHost (true);
         fourBandParam.setValueNotifyingHost (false);
     }
     else if (currentState.first == BandState::ThreeBands)
     {
-        currentState = std::make_pair (BandState::FourBands, 4);
         threeBandParam.setValueNotifyingHost (true);
         fourBandParam.setValueNotifyingHost (true);
     }
     else if (currentState.first == BandState::FourBands)
     {
-        currentState = std::make_pair (BandState::TwoBands, 2);
         threeBandParam.setValueNotifyingHost (false);
         fourBandParam.setValueNotifyingHost (false);
     }
+
+    threeBandAttachment.setValueAsCompleteGesture(threeBandParam.get(), um);
+    fourBandAttachment.setValueAsCompleteGesture(fourBandParam.get(), um);
```

Comment:
So we would want to do this _in place_ of `param.setValueNotifyingHost()`, since `setValueAdCompleteGesture()` will call `setValueNotifyingHost()` internally (as well as taking care of the undo behaviour, and some other fun things).

So it should be something like:
```cpp
if (currentState.first == BandState::TwoBands)
{
    threeBandAttachment.setValueAsCompleteGesture(true, um);
    fourBandAttachment.setValueAsCompleteGesture(false um);
}
```
(and similar for the other if/else cases)